packages/adp-tooling/src/base/change-utils.ts

```diff
@@ -0,0 +1,202 @@
+import type { Editor } from 'mem-fs-editor';
+import path, { posix } from 'path';
+
+import type {
+    AnnotationsData,
+    ChangeTypes,
+    GeneratorName,
+    BaseData,
+    InboundContent,
+    PropertyValueType
+} from '../types';
+import { FolderTypes, AnnotationFileSelectType } from '../types';
+import type { Dirent } from 'fs';
+import { existsSync, readFileSync, readdirSync } from 'fs';
+
+type InboundChangeData = { filePath: string; changeWithInboundId: InboundChange | undefined };
+type InboundChange = { content?: InboundContent };
+
+/**
+ * Writes annotation changes to the specified project path using the provided `mem-fs-editor` instance.
+ *
+ * @param {string} projectPath - The root path of the project.
+ * @param {AnnotationsData} data - The data object containing information about the annotation change.
+ * @param {object} change - The annotation data change that will be written.
+ * @param {Editor} fs - The `mem-fs-editor` instance used for file operations.
+ * @returns {void}
+ */
+export function writeAnnotationChange(projectPath: string, data: AnnotationsData, change: object, fs: Editor): void {
+    try {
+        const { answers, timestamp, annotationFileName } = data;
+        const changeFileName = `id_${timestamp}_addAnnotationsToOData.change`;
+        const changesFolderPath = path.join(projectPath, FolderTypes.WEBAPP, FolderTypes.CHANGES);
+        const manifestFolderPath = path.join(changesFolderPath, FolderTypes.MANIFEST);
+        const annotationsFolderPath = path.join(changesFolderPath, FolderTypes.ANNOTATIONS);
+
+        writeChangeToFile(`${manifestFolderPath}/${changeFileName}`, change, fs);
+
+        if (answers.targetAnnotationFileSelectOption === AnnotationFileSelectType.NewEmptyFile) {
+            fs.write(`${annotationsFolderPath}/${annotationFileName}`, '');
```

Comment:
Please check if forward slashes in the paths will work for different operating systems. I think we had issues with that and it was fixed.

Reply:
I've checked on Windows, with what is pushed till this moment and I can not reproduce the old issue and also don't understand why it was an issue in the first place and how it got to that point that it started generating some strange paths inside the change. Tested all the sub-generators and saw no problem (following the same steps described in the issue). Hopefully, the issue is obsolete.

Reply:
Added in: [83d3688](https://github.com/SAP/open-ux-tools/pull/1643/commits/83d36886cfe672437a7497660bd838323e03d8bb)